File: php/control_flow/goto_instruction/goto_instruction.php
<?php
/*
 * Copyright (C) 2019-2026 Katarzyna Krasińska
 * Languagium - https://github.com/katheroine/languagium
 * Licensed under GPL-3.0 - see LICENSE.md
 */

$c = 10;
$a = 0;

start:
$a += $c;
$c--;
if ($c == 0)
  goto stop;
goto start;

stop:
print($a . "\n");

File: perl/control_flow/goto_instruction/goto_instruction.pl
#
# Copyright (C) 2019-2026 Katarzyna Krasińska
# Languagium - https://github.com/katheroine/languagium
# Licensed under GPL-3.0 - see LICENSE.md
#

use strict;
use warnings;

my $c = 10;
my $a = 0;

START:
$a += $c;
$c--;
if ($c == 0) {
  goto STOP;
}
goto START;

STOP:
print($a . "\n");
